### ISSUE TITLE
Consider reviving our old IDE

### ISSUE THREAD

**Issue by rlafuente** [shoebot/shoebot#138]
Gedit 3 is not optimal anymore to serve as an IDE for Shoebot. Maybe we should go back to our own mini-IDE?

Yesterday I showed Shoebot to some interested people and they indicated that the "killer feature" of graphics coding environments such as Processing, Drawbot or Nodebox is exactly their ready-to-use IDE.

**Comment by stuaxo** [comment]
Definitely do it if you have the time :)

I did most of the work to port it to new versions of Gtk as we went, but I either didn't port the menus or there is a bug that makes them not display.

If you have shoebot installed you can currently run it, and pass in a filename:
```sh
$ python shoebot/gui/ide.py examples/animation/hypnoval.bot
```

But with no menus

![image](https://user-images.githubusercontent.com/179677/39411464-9eeabde4-4c02-11e8-9256-e0013508c8c9.png)

OT: It's also a shame we never got the sublime text plugin someone made at the workshop.

**Comment by rlafuente** [comment]
I'll look into this then :-)

**Comment by rlafuente** [comment]
I think I've made good progress in reimplementing the menu. Right now it seems to be back and functioning as expected. [This tutorial](http://python-gtk-3-tutorial.readthedocs.io/en/latest/menus.html?highlight=menu) was my main reference.

![2018-05-01-125208_1920x1080_scrot](https://user-images.githubusercontent.com/159868/39472038-99d5edfa-4d3e-11e8-9e60-cdc3583dcec0.png)

Quick note for later memory -- I've removed the Var Window toggle since the shoebot window checks if there's any var() calls to determine whether to show the varwindow or not.

There's still a couple of issues and enhancements to make, but maybe we create other issues for those and close this one?

**Comment by stuaxo** [comment]
Yup, makes sense to me - great work, it feels like shoebot is coming full-circle now :)

**Comment by stuaxo** [comment]
Yeah ... I wasn't saying don't do the editor... just that the APIs I made for Gedit should have been ready for you to use :)

**Comment by rlafuente** [comment]
Oh yes, I worded myself badly -- it's that I'd prefer to have something like Gedit2 that we could "turn" into the Shoebot editor, instead of having to maintain one from scratch. Just venting, didn't mean to suggest anything, if anything I'm super grateful for the work you've done on both the editor and the Gedit plugin :-)

**Comment by stuaxo** [comment]
Both venting about the same thing :)
